alibi/explainers/similarity/base.py

```diff
@@ -79,21 +82,42 @@ def fit(self,
         self
             Returns self.
         """
-        self.X_train: np.ndarray = X_train
+        self.X_train: Union[np.ndarray, List[Any]] = X_train
         self.Y_train: np.ndarray = Y_train
```

Comment:
Not sure these type hints are needed here as they haven't changed from the definitions in the `fit` signature? `mypy` should be able to infer that the types are the same.